File: test-suite-python/src/test/python/micronaut/docs/hibernate/graalvm/Order.py
# tag::imports[]
from typing import Annotated

from jakarta.persistence import EmbeddedId, Entity, Table

from micronaut.docs.hibernate.graalvm.OrderId import OrderId
# end::imports[]


# tag::clazz[]
@Entity
@Table(name="orders")
class Order:
    id: Annotated[OrderId | None, EmbeddedId] = None
    customer: str | None = None
# end::clazz[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/hibernate/graalvm/Order.kt
package io.micronaut.docs.hibernate.graalvm

// tag::imports[]
import jakarta.persistence.EmbeddedId
import jakarta.persistence.Entity
import jakarta.persistence.Table
// end::imports[]

// tag::clazz[]
@Entity
@Table(name = "orders")
open class Order {

    @EmbeddedId
    open var id: OrderId? = null

    open var customer: String? = null
}
// end::clazz[]
